File: Rust/Sorts/bubblesort.rs
fn bubblesort(array: &mut [i32]) {
    let mut n = array.len();
    while n > 0 {
        let mut newn = 0;
        for i in 0..n-1 {
            if array[i] > array[i+1] {
                let temp = array[i];
                array[i] = array[i+1];
                array[i+1] = temp;
                newn = i+1;
            }
        }
        n = newn;
    }
}

fn main() {
    let mut values = [5, 8, 4, 1, 7, 2, 3, 6];
    println!("values: {:?}", values);
    bubblesort(&mut values);
    println!("sorted: {:?}", values);
}

File: dart/BubbleSort.dart
import 'dart:core';

void main(){
  List<int> list = [64, 34, 25, 12, 22, 11, 90];

 bubbleSort(list);

  print("Sorted list is:");
  for (var i = 0; i < list.length; i++) {
    print(list[i]);
  }
}

void bubbleSort(List<int> list){
  var n = list.length;
  print(n);
  for (var i = 0; i < n-1; i++) {
    for (var j = 0; j < n-i-1; j++) {
      if(list[j] > list[j+1]){
        var temp = list[j];
        list[j] = list[j+1];
        list[j+1] = temp;
      }
    }
  }
}